File: server/ruby/subscribe-user-to-list.rb
require 'dotenv'
require 'trycourier'

env_path = File.join(File.dirname(__FILE__), '..', '.env')
Dotenv.load(env_path)

api_key = ENV['COURIER_API_KEY']
list_id = ENV['COURIER_SUBSCRIBE_USER_TO_LIST_LIST_ID']
user_id = ENV['COURIER_SUBSCRIBE_USER_TO_LIST_USER_ID']

client = Trycourier::Client.new(api_key: api_key)

client.lists.subscriptions.subscribe_user(user_id, list_id: list_id)

File: server/go/subscribe-user-to-list.go
//go:build ignore

// +build ignore

package main

import (
	"context"
	"fmt"
	"os"
	"path/filepath"

	"github.com/joho/godotenv"
	"github.com/trycourier/courier-go/v4"
	"github.com/trycourier/courier-go/v4/option"
)

func LoadEnv() {
	envPath := filepath.Join("..", ".env")
	godotenv.Load(envPath)
}

func main() {
	LoadEnv()

	apiKey := os.Getenv("COURIER_API_KEY")
	listID := os.Getenv("COURIER_SUBSCRIBE_USER_TO_LIST_LIST_ID")
	userID := os.Getenv("COURIER_SUBSCRIBE_USER_TO_LIST_USER_ID")

	client := courier.NewClient(
		option.WithAPIKey(apiKey),
	)

	requestBody := map[string]interface{}{
		"preferences": map[string]interface{}{
			"categories":    map[string]interface{}{},
			"notifications": map[string]interface{}{},
		},
	}

	var response map[string]interface{}
	err := client.Put(
		context.Background(),
		fmt.Sprintf("/lists/%s/subscriptions/%s", listID, userID),
		requestBody,
		&response,
	)

	if err != nil {
		panic(err.Error())
	}

	fmt.Printf("%+v\n", response)
}
